### PR TITLE
Add NUMA awareness to Lago.

### PR DESCRIPTION
**Pull request by mykaul**
Logic:

If there's >1 vCPU:
2 vCPUs - 2 NUMA nodes, each has 1 vCPU
4 vCPUS - 2 NUMA nodes, each has 1 vCPU

Otherwise:

If num of vCPUs (N) divisible by 4:
N/4 NUMA nodes, each has 4 vCPUs
If num of vCPUs divisible by 2:
N/2 NUMA nodes, each has 2 vCPUs

Otherwise,
N NUMA nodes, each has 1 vCPU.

Tested with 1 - 6 vCPUs.

### CODE REVIEW

**Comment by mykaul** [comment]
Resolves https://github.com/lago-project/lago/issues/694

**Comment by gbenhaim** [comment]
`11:44:13 tests/unit/lago/providers/libvirt/test_cpu.py::TestCPU::test_generate_host_passthrough FAILED [ 94%]`

`11:44:13 tests/unit/lago/providers/libvirt/test_cpu.py::TestCPU::test_init_default FAILED [ 95%]`

**Comment by mykaul** [comment]
Working in changing the tests (unfortunately considerably) to adapt to the new feature...

**Comment by mykaul** [comment]
@gbenhaim , @eedri - any ideas what's causing the failures? I don't see anything wrong, but perhaps I'm missing something (we really should have colored highlights to the log file...)

**Comment by mykaul** [comment]
ci test please

**Comment by eedri** [comment]
It might be related to a recent change in CI, @ifireball and @gbenhaim will check it later today or tomorrow.

File: lago/providers/libvirt/cpu.py [hunk 1]
```diff
@@ -244,6 +247,75 @@ def generate_topology(self, vcpu_num, cores=1, threads=1):
             threads=str(threads),
         )
 
+    def generate_numa(self, vcpu_num):
+        """
+        Generate guest CPU <numa> XML child
+        Configures 1, 2 or 4 vCPUs per cell.
+
+        Args:
+            vcpu_num(int): number of virtual CPUs
```
Comment (gbenhaim) [inline lago/providers/libvirt/cpu.py]: The docstrings don't match the function's signature.

**Comment by gbenhaim** [comment]
@ifireball @danielBelenky Seems like an issue with `pusher.py`

**Comment by ifireball** [comment]
@gbenhaim see: https://ovirt-jira.atlassian.net/browse/OVIRT-1910

**Comment by mykaul** [comment]
ci test please

**Comment by mykaul** [comment]
@gbenhaim , @ifireball - any idea why FC27 seems to be stuck?
BTW, I'm not sure why we are still testing on FC25 (and potentially FC26 as well) - just a waste of resources if you ask me.

**Comment by eedri** [comment]
updating distros: https://gerrit.ovirt.org/#/c/87406/,
I think @gbenhaim wants to leave fc26 for now, we can probably drop it when we add fc28.

**Comment by gbenhaim** [comment]
@mykaul fixed. The fc27 job expected to run on a fc27 slave (we had some issue in the past with virt-sysprep when not restricting the slave distro).

I've manually changed it to run on fc26 slaves and asked ederevea to bring up fc27 slaves.

File: lago/providers/libvirt/cpu.py [hunk 2]
```diff
@@ -244,12 +247,81 @@ def generate_topology(self, vcpu_num, cores=1, threads=1):
             threads=str(threads),
         )
 
+    def generate_numa(self, vcpu_num):
+        """
+        Generate guest CPU <numa> XML child
+        Configures 1, 2 or 4 vCPUs per cell.
+
+        Args:
+            vcpu_num(str): number of virtual CPUs
```
Comment (gbenhaim) [inline lago/providers/libvirt/cpu.py]: Looks good!
Just a fix for the docstrings and we can merge.

Comment (mykaul) [reply lago/providers/libvirt/cpu.py]: I fixed the docstring - I don't like it, but it is a string.

Comment (gbenhaim) [reply lago/providers/libvirt/cpu.py]: I'm talking about:
```
+            cores(int): number of cores
+            threads(int): number of threads
```
They are not part of the function's signature.

Comment (mykaul) [reply lago/providers/libvirt/cpu.py]: Fixed, thanks.

**Comment by gbenhaim** [comment]
ci merge please

**Comment by mykaul** [comment]
Looks like we are still failing on FC25  - see http://jenkins.ovirt.org/job/lago_master_github_check-merged-pipeline/199/console

**Comment by gbenhaim** [comment]
ci merge please

**Pull request merged**